Gem/Code/Source/Components/Multiplayer/PlayerKnockbackEffectComponent.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PlayerKnockbackBus.h>
+#include <Multiplayer/Components/NetworkCharacterComponent.h>
+#include <Multiplayer/Components/NetworkTransformComponent.h>
+#include <Source/Components/Multiplayer/PlayerKnockbackEffectComponent.h>
+
+namespace MultiplayerSample
+{
+    PlayerKnockbackEffectComponentController::PlayerKnockbackEffectComponentController(PlayerKnockbackEffectComponent& parent)
+        : PlayerKnockbackEffectComponentControllerBase(parent)
+    {
+    }
+
+    void PlayerKnockbackEffectComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void PlayerKnockbackEffectComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void PlayerKnockbackEffectComponentController::HandleRPC_Knockback([[maybe_unused]] AzNetworking::IConnection* invokingConnection,
+        const AZ::Vector3& direction)
+    {
+        const AZ::Vector3& position = GetNetworkTransformComponentController()->GetTranslation();
+        PlayerKnockbackNotificationBus::Event(GetEntityId(), &PlayerKnockbackNotificationBus::Events::OnPlayerKnockback, position, direction);
+
+        // Spread out the move to avoid issues in a physical simulation.
+        const float largestPossibleStep = 0.25f;
+        const int steps = aznumeric_cast<int>(direction.GetLength() / largestPossibleStep) + 1;
+        if (steps > 0)
```

Comment:
Nice! Not for this PR, but wondering if TryMove should be improved to handle this; I would have assumed TryMove could work over longer distances.

Reply:
It is not clear if PhysX does or not. It sounds like it does but maybe not https://gameworksdocs.nvidia.com/PhysX/4.0/documentation/PhysXAPI/files/classPxController.html#a1cbd9da6b634f45585df92bb71e37782